core/common/src/main/java/alluxio/conf/Configuration.java

```diff
@@ -329,38 +336,132 @@ public static Map<String, Object> toMap() {
    *         null
    */
   public static Map<String, Object> toMap(ConfigurationValueOptions opts) {
-    return sConf.toMap(opts);
+    return SERVER_CONFIG_REFERENCE.get().toMap(opts);
+  }
+
+  /**
+   * @return the global configuration
+   */
+  public static AlluxioConfiguration global() {
+    return SERVER_CONFIG_REFERENCE.get();
+  }
+
+  /**
+   * @return the global configuration
+   */
+  public static InstancedConfiguration modifiableGlobal() {
+    return SERVER_CONFIG_REFERENCE.get();
   }
 
   /**
-   * @return the {@link InstancedConfiguration} object backing the global configuration
+   * @return a copy of {@link InstancedConfiguration} object based on the global configuration
    */
-  public static InstancedConfiguration global() {
-    return sConf;
+  public static InstancedConfiguration copyGlobal() {
+    return new InstancedConfiguration(SERVER_CONFIG_REFERENCE.get().copyProperties());
   }
 
   /**
    * Loads cluster default values for workers from the meta master if it's not loaded yet.
    *
    * @param address the master address
+   * @param scope the property scope
    */
-  public static synchronized void loadWorkerClusterDefaults(InetSocketAddress address)
+  public static void loadClusterDefaults(InetSocketAddress address, Scope scope)
       throws AlluxioStatusException {
-    if (sConf.getBoolean(PropertyKey.USER_CONF_CLUSTER_DEFAULT_ENABLED)
-        && !sConf.clusterDefaultsLoaded()) {
-      GetConfigurationPResponse response = ConfigurationUtils.loadConfiguration(address, sConf,
-          false, true);
-      AlluxioConfiguration conf = ConfigurationUtils.getClusterConf(response, sConf, Scope.WORKER);
-      sConf = new InstancedConfiguration(conf.copyProperties(), conf.clusterDefaultsLoaded());
+    InstancedConfiguration conf = SERVER_CONFIG_REFERENCE.get();
+    InstancedConfiguration newConf;
+    if (conf.getBoolean(PropertyKey.USER_CONF_CLUSTER_DEFAULT_ENABLED)
+        && !conf.clusterDefaultsLoaded()) {
+      do {
+        GetConfigurationPResponse response = ConfigurationUtils
+            .loadConfiguration(address, conf, false, true);
+        newConf = ConfigurationUtils.getClusterConf(response, conf, scope);
+      } while (!SERVER_CONFIG_REFERENCE.compareAndSet(conf, newConf));
```

Comment:
When two threads loadClusterDefaults, one will keep going? because `conf` is never updated?

Reply:
🤦‍♀️ thanks for catching...

Reply:
Though, I don't think this function will be called from multiple threads. They are only called in AlluxioWorker and AlluxioJobWorker in the main function.